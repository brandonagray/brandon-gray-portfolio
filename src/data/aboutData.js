import React from 'react';

export const STATS_COUNTER_DURATION = 3;

export const STATS_LIST = [
	{
		name: 'Trips Around The Sun',
		start: 0,
		end: 34,
	},
	{
		name: 'Years Coding',
		start: 0,
		end: 12,
	},
	{
		name: 'Countries Visited',
		start: 0,
		end: 10,
	},
];

export const DESC_PARAGRAPHS = [
	<span>
		I love solving problems (which is why I enjoy coding)! But problem solving
		isn't about sitting in a board room talking about the way things should be
		done. I have a strong background in design which allows me to think on my
		feet and act upon it. My skills & experience allow me to come up with
		creative solutions that form perfect harmonies between form and function
		while maintaining the highest levels of aesthetic standard.
	</span>,
	<span>
		I've found that the key to creating great interactive experiences comes from
		understanding what the client and user needs, among other things. The
		essential stage of concept development, planning is at the heart of any
		successful project. I take great care to ensure all the necessary steps are
		taken to produce stellar results every time.
	</span>,
	<span>
		It also takes talent, knowledge, hard work, and dedication to make great
		ideas come to life. With 10+ years of experience now under my belt, I'm not
		afraid of a challenge. New technologies get me excited and I love jumping
		right in and getting things done, untangling problems and learning the
		lessons that unfold with each project.
	</span>,
	<span>
		While my focus is on the front end - designing and developing intuitive,
		stylistic and user friendly web interfaces - I'm extremely interested in the
		workings of the entire web stack. Stemming from the fact that I'm a very
		visual learner, I have a deep fascination with user interfaces and UX
		design. I love trawling the web for interesting articles on stuff like
		human-computer interactions and subtle effects on user experience are my
		bread and butter. It's a major contributing factor to why I also use tools
		like Adobe XD and InVision.
	</span>,
	<span>
		When I'm not in front of a computer, I thoroughly enjoy more hands-on
		creative endeavors - pottery and mosaic making have become a fun pasttime
		for me. Other interests of mine include keeping fit with Brazilian Jiu-Jitsu
		ü§ôüèø, fencing ü§∫, cooking, and travelling (I'm currently on a year long
		working-holiday in New Zealand!).
	</span>,
	<span>
		If you've made it this far, thanks for reading ü•Ç! I'm always looking for
		new opportunities. If you have any questions, feel free to shoot me an
		email, or reach out on social media!
	</span>,
];

export const INTERESTS_LIST = [
	{ name: 'Travel', icon: 'plane' },
	{ name: 'Fencing', icon: 'fencing' },
	{ name: 'Mosaic', icon: 'mosaic' },
	{ name: 'Coding', icon: 'code' },
	{ name: 'Volunteering', icon: 'heart' },
];

export const SKILLS_LIST = [
	{
		type: 'languages',
		children: [
			{ name: 'JavaScript (ES6+)', icon: 'javascript' },
			{ name: 'PHP', icon: 'php' },
			{ name: 'Python', icon: 'python' },
		],
	},
	{
		type: 'frameworks',
		children: [
			{ name: 'Angular', icon: 'angular' },
			{ name: 'React', icon: 'react' },
			{ name: 'HTML', icon: 'html5' },
			{ name: 'CSS', icon: 'css3' },
			{ name: 'Ionic 3', icon: 'ionic' },
		],
	},
	{
		type: 'tools',
		children: [
			{ name: 'Terminal (Bash)', icon: 'terminal' },
			{ name: 'Git', icon: 'git_branch' },
			{ name: 'SQL Databases', icon: 'database' },
			{ name: 'MongoDB', icon: 'mongodb' },
			{ name: 'Adobe XD', icon: 'adobexd' },
			{ name: 'Adobe Photoshop', icon: 'photoshop' },
			{ name: 'Creative Problem Solving', icon: 'stackoverflow' },
		],
	},
];
