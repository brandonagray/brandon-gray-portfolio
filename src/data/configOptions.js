import { css } from 'styled-components';

const displaySizes = {
  desktop: 2160,
  tablet: 1024,
  phone: 425
}

// Iterate through the sizes and create a media template
export const mediaSize = Object.keys(displaySizes).reduce((acc, label) => {
  acc[label] = (...args) => css`
    @media (max-width: ${displaySizes[label] / 16}em) {
      ${css(...args)}
    }
  `
  return acc
}, {})



export const menuPageOptions = [
  {
    text: 'Home',
    route: '/',
    colour: '#85CEE8'
  },
  {
    text: 'About',
    route: '/about',
    colour: '#02A3EE'
  },
  {
    text: 'Work Experience',
    route: '/experience',
    colour: '#65d4FF'
  },
  {
    text: 'Past Projects',
    route: '/projects',
    colour: '#305077'
  },
  {
    text: 'Blog',
    route: '/blog',
    colour: '#00BA93'
  }
];

export const contactOptions = [
  {
    text: 'say hi!',
  },
  {
    text: 'resume'
  },
  {
    text: 'github'
  },
  {
    text: 'linkedin'
  }
];

export const resumeOptions = [
  {
    name: 'Current',
    previewSource: '/img/resume/current-resume.png',
    downloadName: 'Brandon Gray - Resume',
    downloadSource: '/docs/current-resume.pdf',
    color: '#88dbe3'
  },
];

export const greetingOptions = ['Hi!', 'Hey!', 'Heya!', 'Yo!', 'Salut!', '👋🏿 !']

export const roleOptions = ['Web Developer 👨🏿‍💻', 'UI Designer 🎨', 'Avid Traveller 🌍', 'Martial Artist 🥋', 'Trekkie 🖖🏿']

export const particleConfig = {
  "particles": {
    "number": {
      "value": 150,
      "density": {
        "enable": true,
        "value_area": 3500
      }
    },
    "color": {
      "value": ["#85CEE8","#02A3EE","#65d4FF","#305077","#DEE7F9"]
    },
    "shape": {
      "type": "circle",
      "stroke": {
        "width": 0,
        "color": "#000000"
      },
      "polygon": {
        "nb_sides": 5
      },
      "image": {
        "src": "img/github.svg",
        "width": 100,
        "height": 100
      }
    },
    "opacity": {
      "value": 0.5,
      "random": true,
      "anim": {
        "enable": false,
        "speed": 1,
        "opacity_min": 0.1,
        "sync": false
      }
    },
    "size": {
      "value": 4.008530152163807,
      "random": true,
      "anim": {
        "enable": false,
        "speed": 40,
        "size_min": 0.1,
        "sync": false
      }
    },
    "line_linked": {
      "enable": false,
      "distance": 150,
      "color": "#ffffff",
      "opacity": 0.4,
      "width": 1
    },
    "move": {
      "enable": true,
      "speed": 2,
      "direction": "none",
      "random": false,
      "straight": false,
      "out_mode": "out",
      "bounce": false,
      "attract": {
        "enable": false,
        "rotateX": 600,
        "rotateY": 1200
      }
    }
  },
  "interactivity": {
    "detect_on": "canvas",
    "events": {
      "onhover": {
        "enable": false,
        "mode": "repulse"
      },
      "onclick": {
        "enable": false,
        "mode": "push"
      },
      "resize": true
    },
    "modes": {
      "grab": {
        "distance": 400,
        "line_linked": {
          "opacity": 1
        }
      },
      "bubble": {
        "distance": 400,
        "size": 40,
        "duration": 2,
        "opacity": 8,
        "speed": 3
      },
      "repulse": {
        "distance": 200,
        "duration": 0.4
      },
      "push": {
        "particles_nb": 4
      },
      "remove": {
        "particles_nb": 2
      }
    }
  },
  "retina_detect": true
}
