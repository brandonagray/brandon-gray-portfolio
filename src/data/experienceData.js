export const experienceList = [
  {
    name: 'Nokia',
    role: 'Senior Web Software Designer',
    color: '#9CCDA1',
    bgImgSource: '/img/experience/sdwanportal.jpg',
    logoImgSource: '/img/experience/nokia_logo.png',
    desc: 'I was responsible for the design, architecture and development of the client side of this next generation service.',
    techStack: [
      {name: 'Angular', icon: 'devicons devicons-angular', color: '#8E2704' },
      {name: 'TypeScript', icon: 'devicons devicons-javascript', color: '#0062A8' },
      {name: 'Webpack', icon: 'devicons devicons-webplatform', color: '#424242' },
      {name: 'HTML', icon: 'devicons devicons-html5', color: '#EE8407' },
      {name: 'CSS Grid', icon: 'devicons devicons-css3', color: '#4173D5' },
    ],
    actionLinks: [
      
    ]
  },
  {
    name: 'Solarwinds N-able',
    role: 'Associate Web Developer',
    color: '#9CCDA1',
    bgImgSource: '/img/experience/nrc.jpg',
    logoImgSource: '/img/experience/swnable_logo.png',
    desc: 'Designed, developed and maintained the customer portal, the companys main website, as well as various microsites and landing pages.',
    techStack: [
      {name: 'JavaScript', icon: 'devicons devicons-javascript', color: '#0062A8' },
      {name: 'Kentico', icon: 'devicons devicons-kentico', color: '#7C2ACF' },
      {name: 'HTML', icon: 'devicons devicons-html5', color: '#EE8407' },
      {name: 'CSS Grid', icon: 'devicons devicons-css3', color: '#4173D5' },
    ],
    actionLinks: [
      {name: "View on Web", icon: "monitor", url: "https://nrc.n-able.com"}
    ]
  },
  {
    name: 'Gray Interactive',
    role: 'Principal Consultant / Developer',
    color: '#b186c4',
    bgImgSource: '/img/experience/grayinteractive.jpg',
    logoImgSource: '/img/experience/grayint_logo.png',
    desc: 'My digital agency website. The business I use to contract under.',
    techStack: [
      {name: 'JavaScript', icon: 'devicons devicons-javascript', color: '#0062A8' },
      {name: 'Gatsby', icon: 'devicons devicons-opensource', color: '#7C2ACF' },
      {name: 'Wordpress', icon: 'devicons devicons-wordpress', color: '#424242' },
      {name: 'HTML', icon: 'devicons devicons-html5', color: '#EE8407' },
      {name: 'CSS Grid', icon: 'devicons devicons-css3', color: '#4173D5' },
    ],
    actionLinks: [
      {name: "View on Web", icon: "monitor", url: "https://www.grayinteractive.co"},
    ]
  },
]
