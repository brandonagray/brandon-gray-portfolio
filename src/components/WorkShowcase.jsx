import React from 'react';
import styled, { css } from 'styled-components';
import VisibilitySensor from 'react-visibility-sensor';
import { isMobile, isIOS } from 'react-device-detect';
import { mediaSize } from '../data/configOptions';

import Icon from './Icon';
import FloatText from './FloatText';

const Container = styled.div`
  position: relative;
  margin-left: auto;
  margin-right: auto;
  width: ${props => (props.expanded ? '90%' : '75%')};
  height: auto;
  margin-bottom: 10vh;
  cursor: pointer;
  color: #F6F6F6;
  border-radius: 5px;

  display: grid;
  padding: 0 5% 0 5%;
  grid-template-rows: auto;
  grid-template-areas: "title"
                       "role"
                       "desc"
                       "indicator";

  background: url(${props => props.bg}) center/cover;
  filter: grayscale(50%);
  transition: all 0.2s ease-in;
  transition: width 0.75s ease, height 0.75s ease;

  /* Pseudo-element for shadow on container during focus */
  &:before {
    content: ' ';
    position: absolute;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;

    box-shadow: 0 10px 50px 0 rgba(0, 0, 0, 0.6);
    border-radius: 5px;

    opacity: 0;
    transition: opacity 500ms;
  }

  &:after {
    content: ' ';
    position: absolute;
    z-index: 5;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    border-radius: 5px;
    background-color: rgba(0, 0, 0, 0.6);
    opacity: 1;
    transition: opacity 500ms;
  }

  ${props =>
    props.focused || props.expanded
      ? css`
          filter: none;
          background: url(${props => props.bg}) center/cover;

          &:before {
            opacity: 1;
          }

          &:after {
            opacity: 0.75;
          }

          ${mediaSize.phone`
      &:after {
        opacity: 0.85;
      }
    `};
        `
      : null}

  ${mediaSize.tablet`
    width: 85%;
    border-radius: 8px;
    &:before, &:after {
      border-radius: 8px;
    }
  `}

  ${mediaSize.phone`
    height: auto;
    width: 80%;
    padding: 0 10% 0 10%;
    margin-bottom: 4em;

    grid-template-rows: auto auto auto;
    grid-template-areas: "title"
                         "role"
                         "desc"
                         "indicator";
  `}

`;

const WorkTitle = styled.span`
  font-size: 4vw;
  font-weight: bold;
  position: relative;
  z-index: 6;
  bottom: 0;
  grid-area: title;
  white-space: nowrap;
  padding-top: 1.5em;
  padding-bottom: 2px;

  ${mediaSize.tablet`
    padding-top: 15vmin;
    font-size: 2.5em;
    letter-spacing: ${isIOS ? '-0.05em' : 0};
  `} ${mediaSize.phone`
    font-size: 2.5em;
    position: relative;
    padding-top: 40vw;
    white-space: normal;
  `};
`;

const WorkLogo = styled.img`
  position: absolute;
  z-index: 6;
  top: -5vh;
  right: -5vw;
  max-width: 25vw;
  max-height: 25vh;

  opacity: 0;
  transition: all 0.3s ease-in;

  ${props =>
    props.focused || props.expanded
      ? css`
          opacity: 1;
        `
      : null}

  ${mediaSize.tablet`
    top: -5vw;
    max-width: 20vmin;
    max-height: 25vmin;
  `}

  ${mediaSize.phone`
    top: -5vw;
    right: -8vw;
    max-width: 43vw;
    max-height: 60vw;
  `}
`;

const WorkRole = styled.div`
  grid-area: role;
  position: relative;
  z-index: 6;
  font-size: 2vw;
  padding-bottom: 1.5em;

  ${mediaSize.tablet`
    font-size: 2em;
    padding-bottom: 1.5em;
  `} ${mediaSize.phone`
    font-size: 5vw;
  `};
`;

const WorkDesc = styled.div`
  grid-area: desc;
  align-self: start;
  padding-bottom: ${props => (props.expanded ? '2em' : 0)};
  line-height: 1.5;
  font-size: 2.5vmin;
  z-index: 6;
  max-width: 100%;
  max-height: ${props => (props.expanded ? '7em' : 0)};
  opacity: ${props => (props.expanded ? 1 : 0)};
  transition: max-height 1s, opacity 0.7s, padding 1s;

  ${mediaSize.tablet`
    max-height: ${props => (props.expanded ? '20em' : 0)};
    padding-bottom: ${props => (props.expanded ? '3vw' : 0)};
  `} ${mediaSize.phone`
    font-size: 1em;
    max-height: ${props => (props.expanded ? '30em' : 0)};
    padding-bottom: ${props => (props.expanded ? '7vw' : 0)};
  `};
`;

const DropdownArrow = styled.span`
  grid-area: indicator;
  z-index: 6;
  align-self: center;
  justify-self: center;
  margin-bottom: 1vw;
`;

class WorkShowcase extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      focused: false,
      expanded: false
    };
  }

  handleFocus(focused) {
    this.setState({ focused });
  }

  render() {
    return (
      <VisibilitySensor
        onChange={isVisible => this.handleFocus(isMobile && isVisible)}
      >
        <Container
          focused={this.state.focused}
          expanded={this.state.expanded}
          bg={this.props.work.bgImgSource}
          onMouseEnter={() => this.handleFocus(true)}
          onMouseLeave={() => this.handleFocus(false)}
          onClick={() =>
            this.setState(prevState => {
              return { expanded: !prevState.expanded };
            })
          }
        >
          <WorkTitle>{this.props.work.name}</WorkTitle>
          <WorkRole>{this.props.work.role}</WorkRole>
          <WorkLogo
            src={this.props.work.logoImgSource}
            focused={this.state.focused}
            expanded={this.state.expanded}
          />
          <WorkDesc expanded={this.state.expanded}>
            {this.props.work.desc}
          </WorkDesc>
          <DropdownArrow>
            <FloatText from={-3} to={0}>
              <Icon
                name={this.state.expanded ? 'chevronUp' : 'chevronDown'}
                size="3vh"
                color="#FFFFFF"
              />
            </FloatText>
          </DropdownArrow>
        </Container>
      </VisibilitySensor>
    );
  }
}

export default WorkShowcase;
