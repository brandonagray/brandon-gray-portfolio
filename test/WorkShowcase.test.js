import React from 'react';
import WorkShowcase from '../src/components/WorkShowcase';
import { experienceList } from '../src/data/experienceData';

describe('WorkShowcase component', () => {
  it('should render the work showcase', () => {
    const experience = render(<WorkShowcase work={experienceList[0]} />);

    expect(experience).toMatchSnapshot();
  });

  it('should show description on click', () => {
    const experience = mount(<WorkShowcase work={experienceList[0]} />);

    expect(experience).toHaveStyleRule('width', '75%');
    experience
      .find('div')
      .at(0)
      .simulate('click');
    expect(experience).toHaveStyleRule('width', '90%');
    expect(experience).toMatchSnapshot();
  });
});
