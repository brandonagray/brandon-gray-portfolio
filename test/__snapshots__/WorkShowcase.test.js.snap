// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkShowcase component should render the work showcase 1`] = `
.c7 svg {
  width: 3vh;
  height: 3vh;
  stroke: #FFFFFF;
}

.c6 {
  -webkit-animation-name: dTtlGn;
  animation-name: dTtlGn;
  -webkit-animation-duration: 3s;
  animation-duration: 3s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
}

.c0 {
  position: relative;
  margin-left: auto;
  margin-right: auto;
  width: 75%;
  height: auto;
  margin-bottom: 10vh;
  cursor: pointer;
  color: #F6F6F6;
  border-radius: 5px;
  display: grid;
  padding: 0 5% 0 5%;
  grid-template-rows: auto;
  grid-template-areas: "title" "role" "desc" "indicator";
  background: url(/img/experience/htn.jpg) center/cover;
  -webkit-filter: grayscale(50%);
  filter: grayscale(50%);
  -webkit-transition: all 0.2s ease-in;
  transition: all 0.2s ease-in;
  -webkit-transition: width 0.75s ease,height 0.75s ease;
  transition: width 0.75s ease,height 0.75s ease;
}

.c0:before {
  content: ' ';
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  box-shadow: 0 10px 50px 0 rgba(0,0,0,0.6);
  border-radius: 5px;
  opacity: 0;
  -webkit-transition: opacity 500ms;
  transition: opacity 500ms;
}

.c0:after {
  content: ' ';
  position: absolute;
  z-index: 5;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: 5px;
  background-color: rgba(0,0,0,0.6);
  opacity: 1;
  -webkit-transition: opacity 500ms;
  transition: opacity 500ms;
}

.c1 {
  font-size: 5vw;
  font-weight: bold;
  position: relative;
  z-index: 6;
  bottom: 0;
  grid-area: title;
  white-space: nowrap;
  padding-top: 1em;
  padding-bottom: 2px;
}

.c3 {
  position: absolute;
  z-index: 6;
  top: -5vh;
  right: -5vw;
  max-width: 40vw;
  max-height: 25vh;
  opacity: 0;
  -webkit-transition: all 0.3s ease-in;
  transition: all 0.3s ease-in;
}

.c2 {
  grid-area: role;
  position: relative;
  z-index: 6;
  font-size: 2vw;
  padding-bottom: 1.5em;
}

.c4 {
  grid-area: desc;
  -webkit-align-self: start;
  -ms-flex-item-align: start;
  align-self: start;
  padding-bottom: 0;
  line-height: 1.5;
  font-size: 2.5vmin;
  z-index: 6;
  max-width: 100%;
  max-height: 0;
  opacity: 0;
  -webkit-transition: max-height 1s,opacity 0.7s,padding 1s;
  transition: max-height 1s,opacity 0.7s,padding 1s;
}

.c5 {
  grid-area: indicator;
  z-index: 6;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  justify-self: center;
  margin-bottom: 1vw;
}

@media (max-width:64em) {
  .c0 {
    width: 85%;
    border-radius: 8px;
  }

  .c0:before,
  .c0:after {
    border-radius: 8px;
  }
}

@media (max-width:26.5625em) {
  .c0 {
    height: auto;
    width: 80%;
    padding: 0 10% 0 10%;
    margin-bottom: 4em;
    grid-template-rows: auto auto auto;
    grid-template-areas: "title" "role" "desc" "indicator";
  }
}

@media (max-width:64em) {
  .c1 {
    padding-top: 15vmin;
    font-size: 4.5em;
    -webkit-letter-spacing: 0;
    -moz-letter-spacing: 0;
    -ms-letter-spacing: 0;
    letter-spacing: 0;
  }
}

@media (max-width:26.5625em) {
  .c1 {
    font-size: 2.5em;
    position: relative;
    padding-top: 40vw;
  }
}

@media (max-width:64em) {
  .c3 {
    top: -5vw;
    max-width: 30vmin;
    max-height: 25vmin;
  }
}

@media (max-width:26.5625em) {
  .c3 {
    top: -5vw;
    right: -8vw;
    max-width: 43vw;
    max-height: 60vw;
  }
}

@media (max-width:64em) {
  .c2 {
    font-size: 2em;
    padding-bottom: 1.5em;
  }
}

@media (max-width:26.5625em) {
  .c2 {
    font-size: 5vw;
  }
}

@media (max-width:64em) {
  .c4 {
    max-height: 0;
    padding-bottom: 0;
  }
}

@media (max-width:26.5625em) {
  .c4 {
    font-size: 1em;
    max-height: 0;
    padding-bottom: 0;
  }
}

<div
  class="c0"
>
  <span
    class="c1"
  >
    Hack the North
  </span>
  <div
    class="c2"
  >
    Frontend Developer
  </div>
  <img
    class="c3"
    src="/img/experience/htn_logo.png"
  />
  <div
    class="c4"
  >
    I'm working as a developer to create engaging user experiences and interfaces for Canada's largest hackathon with over 1000 attendees and thousands more applicants and users.
  </div>
  <span
    class="c5"
  >
    <div
      class="c6"
      from="-3"
      to="0"
    >
      <span
        class="c7"
        color="#FFFFFF"
      >
        <svg
          fill="none"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
        >
          <path
            d="M6 9l6 6 6-6"
          />
        </svg>
      </span>
    </div>
  </span>
</div>
`;

exports[`WorkShowcase component should show description on click 1`] = `
.c7 svg {
  width: 3vh;
  height: 3vh;
  stroke: #FFFFFF;
}

.c6 {
  -webkit-animation-name: dTtlGn;
  animation-name: dTtlGn;
  -webkit-animation-duration: 3s;
  animation-duration: 3s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
}

.c0 {
  position: relative;
  margin-left: auto;
  margin-right: auto;
  width: 90%;
  height: auto;
  margin-bottom: 10vh;
  cursor: pointer;
  color: #F6F6F6;
  border-radius: 5px;
  display: grid;
  padding: 0 5% 0 5%;
  grid-template-rows: auto;
  grid-template-areas: "title" "role" "desc" "indicator";
  background: url(/img/experience/htn.jpg) center/cover;
  -webkit-filter: grayscale(50%);
  filter: grayscale(50%);
  -webkit-transition: all 0.2s ease-in;
  transition: all 0.2s ease-in;
  -webkit-transition: width 0.75s ease,height 0.75s ease;
  transition: width 0.75s ease,height 0.75s ease;
  -webkit-filter: none;
  filter: none;
  background: url(/img/experience/htn.jpg) center/cover;
}

.c0:before {
  content: ' ';
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  box-shadow: 0 10px 50px 0 rgba(0,0,0,0.6);
  border-radius: 5px;
  opacity: 0;
  -webkit-transition: opacity 500ms;
  transition: opacity 500ms;
}

.c0:after {
  content: ' ';
  position: absolute;
  z-index: 5;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: 5px;
  background-color: rgba(0,0,0,0.6);
  opacity: 1;
  -webkit-transition: opacity 500ms;
  transition: opacity 500ms;
}

.c0:before {
  opacity: 1;
}

.c0:after {
  opacity: 0.75;
}

.c1 {
  font-size: 5vw;
  font-weight: bold;
  position: relative;
  z-index: 6;
  bottom: 0;
  grid-area: title;
  white-space: nowrap;
  padding-top: 1em;
  padding-bottom: 2px;
}

.c3 {
  position: absolute;
  z-index: 6;
  top: -5vh;
  right: -5vw;
  max-width: 40vw;
  max-height: 25vh;
  opacity: 0;
  -webkit-transition: all 0.3s ease-in;
  transition: all 0.3s ease-in;
  opacity: 1;
}

.c2 {
  grid-area: role;
  position: relative;
  z-index: 6;
  font-size: 2vw;
  padding-bottom: 1.5em;
}

.c4 {
  grid-area: desc;
  -webkit-align-self: start;
  -ms-flex-item-align: start;
  align-self: start;
  padding-bottom: 2em;
  line-height: 1.5;
  font-size: 2.5vmin;
  z-index: 6;
  max-width: 100%;
  max-height: 7em;
  opacity: 1;
  -webkit-transition: max-height 1s,opacity 0.7s,padding 1s;
  transition: max-height 1s,opacity 0.7s,padding 1s;
}

.c5 {
  grid-area: indicator;
  z-index: 6;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  justify-self: center;
  margin-bottom: 1vw;
}

@media (max-width:26.5625em) {
  .c0:after {
    opacity: 0.85;
  }
}

@media (max-width:64em) {
  .c0 {
    width: 85%;
    border-radius: 8px;
  }

  .c0:before,
  .c0:after {
    border-radius: 8px;
  }
}

@media (max-width:26.5625em) {
  .c0 {
    height: auto;
    width: 80%;
    padding: 0 10% 0 10%;
    margin-bottom: 4em;
    grid-template-rows: auto auto auto;
    grid-template-areas: "title" "role" "desc" "indicator";
  }
}

@media (max-width:64em) {
  .c1 {
    padding-top: 15vmin;
    font-size: 4.5em;
    -webkit-letter-spacing: 0;
    -moz-letter-spacing: 0;
    -ms-letter-spacing: 0;
    letter-spacing: 0;
  }
}

@media (max-width:26.5625em) {
  .c1 {
    font-size: 2.5em;
    position: relative;
    padding-top: 40vw;
  }
}

@media (max-width:64em) {
  .c3 {
    top: -5vw;
    max-width: 30vmin;
    max-height: 25vmin;
  }
}

@media (max-width:26.5625em) {
  .c3 {
    top: -5vw;
    right: -8vw;
    max-width: 43vw;
    max-height: 60vw;
  }
}

@media (max-width:64em) {
  .c2 {
    font-size: 2em;
    padding-bottom: 1.5em;
  }
}

@media (max-width:26.5625em) {
  .c2 {
    font-size: 5vw;
  }
}

@media (max-width:64em) {
  .c4 {
    max-height: 20em;
    padding-bottom: 3vw;
  }
}

@media (max-width:26.5625em) {
  .c4 {
    font-size: 1em;
    max-height: 30em;
    padding-bottom: 7vw;
  }
}

<WorkShowcase
  work={
    Object {
      "bgImgSource": "/img/experience/htn.jpg",
      "color": "#7bbec1",
      "desc": "I'm working as a developer to create engaging user experiences and interfaces for Canada's largest hackathon with over 1000 attendees and thousands more applicants and users.",
      "logoImgSource": "/img/experience/htn_logo.png",
      "name": "Hack the North",
      "role": "Frontend Developer",
    }
  }
>
  <VisibilitySensor
    active={true}
    containment={null}
    delayedCall={false}
    intervalCheck={true}
    intervalDelay={100}
    minTopValue={0}
    offset={Object {}}
    onChange={[Function]}
    partialVisibility={false}
    resizeCheck={false}
    resizeDelay={250}
    resizeThrottle={-1}
    scrollCheck={false}
    scrollDelay={250}
    scrollThrottle={-1}
  >
    <styled.div
      bg="/img/experience/htn.jpg"
      expanded={true}
      focused={false}
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="c0"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <styled.span>
          <span
            className="c1"
          >
            Hack the North
          </span>
        </styled.span>
        <styled.div>
          <div
            className="c2"
          >
            Frontend Developer
          </div>
        </styled.div>
        <styled.img
          expanded={true}
          focused={false}
          src="/img/experience/htn_logo.png"
        >
          <img
            className="c3"
            src="/img/experience/htn_logo.png"
          />
        </styled.img>
        <styled.div
          expanded={true}
        >
          <div
            className="c4"
          >
            I'm working as a developer to create engaging user experiences and interfaces for Canada's largest hackathon with over 1000 attendees and thousands more applicants and users.
          </div>
        </styled.div>
        <styled.span>
          <span
            className="c5"
          >
            <FloatText
              from={-3}
              to={0}
            >
              <styled.div
                from={-3}
                to={0}
              >
                <div
                  className="c6"
                  from={-3}
                  to={0}
                >
                  <Icon
                    color="#FFFFFF"
                    name="chevronUp"
                    size="3vh"
                  >
                    <styled.span
                      color="#FFFFFF"
                      size="3vh"
                    >
                      <span
                        className="c7"
                        color="#FFFFFF"
                        size="3vh"
                      >
                        <svg
                          fill="none"
                          strokeLinecap="round"
                          strokeLinejoin="round"
                          strokeWidth="2"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M18 15l-6-6-6 6"
                          />
                        </svg>
                      </span>
                    </styled.span>
                  </Icon>
                </div>
              </styled.div>
            </FloatText>
          </span>
        </styled.span>
      </div>
    </styled.div>
  </VisibilitySensor>
</WorkShowcase>
`;
